server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1397,6 +1377,63 @@ public boolean performConsumerUpdates(ConsumerDTO updated, Consumer toUpdate,
         return changesMade;
     }
 
+    private boolean updateEnvironment(ConsumerDTO updated, Consumer toUpdate) {
+        boolean changesMade = false;
+        boolean environmentFound = false;
+        Environment e = null;
+        String environmentId = updated.getEnvironment() == null ? null : updated.getEnvironment().getId();
+        String environmentName = updated.getEnvironment() == null ? null : updated.getEnvironment().getName();
+
+        if (environmentName != null && environmentId == null) {
+            List<Environment> environment = environmentCurator.listForOwnerByName(toUpdate.getOwner(),
+                environmentName).list();
+
+            if (!environment.isEmpty() && environment.size() == 1) {
+                e = environment.get(0);
+                environmentId = environment.get(0).getId();
+                environmentFound = true;
+            }
+            else {
+                throw new NotFoundException(i18n.tr(
+                        "Environment with name \"{0}\" could not be found.", environmentName));
+            }
+
+        }
+
+        if (environmentId != null && (toUpdate.getEnvironmentId() == null ||
+            !toUpdate.getEnvironmentId().equals(environmentId))) {
+
+            if (!environmentFound) {
+                e = environmentCurator.get(environmentId);
+
+                if (e == null) {
+                    throw new NotFoundException(i18n.tr(
+                            "Environment with ID \"{0}\" could not be found.", environmentId));
+                }
+            }
+
+            changesMade = true;
+
+            log.info("Updating environment to: {}", environmentId);
+        }
+
+        if (e != null && changesMade) {
```

Comment:
I don't quite follow the intent behind changesMade here. It's set only in one of the lookup cases above, but it's not associated with any actual change. This seems like something that should be set only if an environment was found; but since we throw an exception in both lookup cases, this only serves the purpose of indicating if (a) no environment reference was provided or (b) a reference was provided *and* a change was made as a result. Following that, this should only be set in the block it's currently guarding.

Looking at the original code, that seems to be what it was doing before.

Reply:
I removed `e != null` check. It seems reasonable that we update only when changesMade flag is true.